ApplicationInsightsInternalLogger/src/main/java/com/microsoft/applicationinsights/internal/logger/InternalLogger.java

```diff
@@ -68,7 +81,12 @@ public int getValue() {
         FILE
     }
 
-    private boolean initialized = false;
+    private volatile boolean initialized = false;
```

Comment:
I think it's good to turn this to AtomicBoolean and same for the props variable. Then we don't need the sync block.

Reply:
AtomicBoolean won't work here because both `initialized` and `propsNotFound` need to be guarded by the same lock

Reply:
Ok in this case wouldn't it be good to just take an explicit lock and guard this block of code? Why do we need to lock on the whole method. Would that be possible?

Reply:
Since the initialize method was synchronized, I kept the new part synchronized as well. I'm not changing the nature of this class, just adding functionality.